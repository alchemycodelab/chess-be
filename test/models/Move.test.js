require('dotenv').config();
const mongoose = require('mongoose');
const connect = require('../../lib/utils/connect');
const Move = require('../../lib/models/Move');

describe('Move test', () => {
  beforeEach(() => {
    return connect();
  });

  it('can not create a move given an invalid chess move', () => {
    const move = new Move({
      game: mongoose.Types.ObjectId(),
      from: 'e2',
      to: 'e5'
    });

    return move
      .save()
      .then(() => {
        // we should never be here
        expect(false).toBeTruthy();
      })
      .catch(err => {
        expect(err.message).toEqual('Invalid Move');
      });
  });

  it('creates a move given a valid chess move', () => {
    const move = new Move({
      game: mongoose.Types.ObjectId(),
      from: 'e2',
      to: 'e4'
    });

    return move
      .save()
      .then(m => {
        expect(m._doc).toMatchSnapshot({
          '_id': expect.any(Object),
          'game': expect.any(Object),
        });
      });
  });
});
