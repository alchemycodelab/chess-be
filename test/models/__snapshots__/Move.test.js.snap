// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Move test creates a move given a valid chess move 1`] = `
Object {
  "__v": 0,
  "_id": Any<Object>,
  "from": "e2",
  "game": Any<Object>,
  "result": "Play-On",
  "to": "e4",
}
`;
